.github/workflows/build-and-deploy.yml

```diff
@@ -1,50 +1,85 @@
-# See https://docs.docker.com/build/ci/github-actions/multi-platform/
+# See https://docs.docker.com/build/ci/github-actions/multi-platform/#distribute-build-across-multiple-runners
 name: Build, Push, and Deploy Image
 
-on:
-  push:
-    branches:
-      - main
-    tags:
-      - v*
-
-permissions:
```

Comment:
Should we keep the workflow-level restrictions so that we don't risk future jobs using default permissions?